trin-core/src/portalnet/overlay_service.rs

```diff
@@ -967,17 +997,46 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
                 "[{:?}] Skipping processing for content connection ID {}",
                 self.protocol, id
             ),
-            Content::Content(_) => {
-                debug!(
-                    "[{:?}] Skipping processing for content bytes",
-                    self.protocol
-                )
-            }
+            Content::Content(content) => self.process_received_content(content, request).await,
             Content::Enrs(enrs) => self
                 .process_discovered_enrs(enrs.into_iter().map(|ssz_enr| ssz_enr.into()).collect()),
         }
     }
 
+    async fn process_received_content(&mut self, content: ByteList, request: FindContent) {
+        let content_key = match TContentKey::try_from(request.content_key) {
+            Ok(val) => val,
+            Err(msg) => {
+                error!("Unable to process received content: Invalid content key: {msg:?}");
+                return;
+            }
+        };
+        if let Err(err) = self
+            .validator
+            .validate_content(&content_key, &content)
+            .await
+        {
+            error!("Unable to validate received content: {err:?}");
+            return;
+        };
+        match self.storage.read().should_store(&content_key) {
+            Ok(should_store) => {
+                if should_store {
+                    if let Err(err) = self.storage.write().store(&content_key, &content.into()) {
```

Comment:
Since validation might be expensive (literally, at first), maybe we should be validating after testing `should_store()`, but obviously before `store()`.